src/Microsoft.Extensions.PlatformAbstractions.Internal/Native/PlatformApis.cs

```diff
@@ -0,0 +1,169 @@
+﻿using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using Microsoft.Extensions.PlatformAbstractions.Internal;
+
+namespace Microsoft.Extensions.PlatformAbstractions.Internal.Native
+{
+    internal static class PlatformApis
+    {
+        private class DistroInfo
+        {
+            public string Id;
+            public string VersionId;
+        }
+
+        private static readonly Lazy<Platform> _platform = new Lazy<Platform>(DetermineOSPlatform);
+        private static readonly Lazy<DistroInfo> _distroInfo = new Lazy<DistroInfo>(LoadDistroInfo);
+
+        public static string GetOSName()
+        {
+            switch (GetOSPlatform())
+            {
+                case Platform.Windows:
+                    return "Windows";
+                case Platform.Linux:
+                    return GetDistroId() ?? "Linux";
+                case Platform.Darwin:
+                    return "Mac OS X";
+                default:
+                    return "Unknown";
+            }
+        }
+
+        public static string GetOSVersion()
```

Comment:
I think a lot of these method could be replaced with `System.Runtime.InteropServices.RuntimeInformation`